crates/wac-resolver/src/registry.rs

```diff
@@ -82,242 +68,120 @@ impl RegistryPackageResolver {
         &self,
         keys: &IndexMap<BorrowedPackageKey<'a>, SourceSpan>,
     ) -> Result<IndexMap<BorrowedPackageKey<'a>, Vec<u8>>, Error> {
-        // Start by fetching any required package logs
-        self.fetch(keys).await?;
-
-        // All the logs have been updated, now we need to see what content
-        // is missing from local storage.
-        let mut packages = IndexMap::new();
-        let missing = self.find_missing_content(keys, &mut packages).await?;
+        // parses into `PackageName` and maps back to `SourceSpan`
+        let package_names_with_source_span = keys
+            .iter()
+            .map(|(key, span)| {
+                Ok((
+                    PackageName::new(key.name.to_string()).map_err(|_| {
+                        Error::InvalidPackageName {
+                            name: key.name.to_string(),
+                            span: *span,
+                        }
+                    })?,
+                    (key.version.cloned(), *span),
+                ))
+            })
+            .collect::<Result<IndexMap<PackageName, (Option<Version>, SourceSpan)>, Error>>()?;
+
+        // fetch required package logs and return error if any not found
+        if let Some(bar) = self.bar.as_ref() {
+            bar.init(1);
```

Comment:
I'm not familiar with how the progress bar works, but I'm surprised to see this hardcoded to `1`. Do we need a progress bar if there's only one thing to do?

Reply:
Fixed. It was left over from the previous behavior.